kernel/task_struct/src/lib.rs

```diff
@@ -485,6 +485,18 @@ impl Task {
         }
     }
 
+    /// Same as [`block`], but doesn't print a warning if the task is already
+    /// blocked.
+    ///
+    /// This method can be useful if blocking potentially blocked tasks in a
+    /// loop (e.g. `test_scheduler`). Logging is very slow, and this function
+    /// can lead to a `100x` performance improvement.
+    pub fn block_no_log(&self) {
```

Comment:
for this and the related `unblock_no_log`, let's just remove the log statements entirely from the existing `block`/`unblock` functions. We have since determined that it is entirely valid to unblock an already runnable task, e.g., in the case of a deferred interrupt handler where the deferred task is already running when another interrupt occurs.